src/utils/db/db.ts

```diff
@@ -591,22 +591,40 @@ export class DB {
     }
 
     public async moveDistinctIds(source: Person, target: Person): Promise<void> {
-        const movedDistinctIdResult = await this.postgresQuery(
-            `
-                UPDATE posthog_persondistinctid
-                SET person_id = $1
-                WHERE person_id = $2
-                  AND team_id = $3
-                RETURNING *
-            `,
-            [target.id, source.id, target.team_id],
-            'updateDistinctIdPerson'
-        )
+        let operationFailedDueToRaceCondition = false
+        let movedDistinctIdResult: QueryResult<any> | null = null
+        try {
+            movedDistinctIdResult = await this.postgresQuery(
+                `
+                    UPDATE posthog_persondistinctid
+                    SET person_id = $1
+                    WHERE person_id = $2
+                      AND team_id = $3
+                    RETURNING *
+                `,
+                [target.id, source.id, target.team_id],
+                'updateDistinctIdPerson'
+            )
+        } catch (error) {
+            if (
+                error.message.includes(
+                    'insert or update on table "posthog_persondistinctid" violates foreign key constraint'
+                )
+            ) {
+                operationFailedDueToRaceCondition = true
+            } else {
+                throw error
+            }
+        }
+
+        // this should never (?) happen, but needed to satisfy the TS compiler
```

Comment:
Let's instead add a type-level assert - this is noise.

You can do this by appending `!` at the end of every `movedDistinctIdResult`

Reply:
Yeah I'm aware of this, was just wondering what would be cleaner. But you're right.